tuple.go

```diff
@@ -133,38 +130,29 @@ func (ipt *IPTuple) unmarshal(ad *netlink.AttributeDecoder) error {
 
 // marshal marshals an IPTuple to a netfilter.Attribute.
 func (ipt IPTuple) marshal() (netfilter.Attribute, error) {
-	// If either address is not a valid IP or if they do not belong to the same address family, returns false.
-	// Taken from net.IP, for some reason this function is not exported.
-	matchAddrFamily := func(ip net.IP, x net.IP) bool {
-		return ip.To4() != nil && x.To4() != nil || ip.To16() != nil && ip.To4() == nil && x.To16() != nil && x.To4() == nil
-	}
-
-	// Ensure that source and destination belong to the same address family.
-	if !matchAddrFamily(ipt.SourceAddress, ipt.DestinationAddress) {
+	if !ipt.SourceAddress.IsValid() || !ipt.DestinationAddress.IsValid() {
 		return netfilter.Attribute{}, errBadIPTuple
 	}
 
 	nfa := netfilter.Attribute{Type: uint16(ctaTupleIP), Nested: true, Children: make([]netfilter.Attribute, 2)}
 
-	// To4() returns nil if the IP is not a 4-byte array nor a 16-byte array with markers
-	// To4() will always return a 4-byte array. To16() will always return a 16-byte array, potentially with markers.
-	// In the case below, To16 can never return markers, because the 4-byte case is caught by To4().
-	if src, dest := ipt.SourceAddress.To4(), ipt.DestinationAddress.To4(); src != nil && dest != nil {
-		nfa.Children[0] = netfilter.Attribute{Type: uint16(ctaIPv4Src), Data: src}
-		nfa.Children[1] = netfilter.Attribute{Type: uint16(ctaIPv4Dst), Data: dest}
+	if ipt.SourceAddress.Is4() && ipt.DestinationAddress.Is4() {
```

Comment:
This would be better expressed as a switch statement now we can use a sane API to detect the address family.